geemap/ee_tile_layers.py

```diff
@@ -151,3 +153,79 @@ def __init__(
             max_zoom=24,
             **kwargs,
         )
+
+    @lru_cache()
+    def _calculate_vis_stats(self, *, bounds, bands):
+        """Calculate stats used for visualization parameters.
+
+        Stats are calculated consistently with the Code Editor visualization parameters,
+        and are cached to avoid recomputing for the same bounds and bands.
+
+        Args:
+            bounds (ee.Geometry|ee.Feature|ee.FeatureCollection): The bounds to sample.
+            bands (tuple): The bands to sample.
+
+        Returns:
+            tuple: The minimum, maximum, standard deviation, and mean values across the
+                specified bands.
+        """
+        stat_reducer = (ee.Reducer.minMax()
+                        .combine(ee.Reducer.mean().unweighted(), sharedInputs=True)
+                        .combine(ee.Reducer.stdDev(), sharedInputs=True))
+
+        stats = self._ee_object.select(bands).reduceRegion(
+            reducer=stat_reducer,
+            geometry=bounds,
+            bestEffort=True,
+            maxPixels=10_000,
+            crs="SR-ORG:6627",
+            scale=1,
```

Comment:
Good question! I went with `scale=1` and the weird CRS because they're used by the Code Editor for calculating vis params (screenshot of the relevant obfuscated code from `playground.js` below). I'm guessing a fixed scale was used to avoid tricky corner cases like images with different projections per band or images with zero bands, which would break if you tried to use `img.projection().nominalScale()` and `img.select(0).projection().nominalScale()`, respectively. I'm open to other ideas though!

![image](https://github.com/gee-community/geemap/assets/50475791/eb151f16-ae89-4230-a371-7d1af0370733)